src/core/markdown.js

```diff
@@ -53,12 +53,48 @@ const gtEntity = /&gt;/gm;
 const ampEntity = /&amp;/gm;
 
 class Renderer extends marked.Renderer {
-  code(code, language, isEscaped) {
+  code(code, infoString, isEscaped) {
+    const { language, ...metaData } = Renderer.parseInfoString(infoString);
+
     // regex to check whether the language is webidl
     if (/(^webidl$)/i.test(language)) {
       return `<pre class="idl">${code}</pre>`;
     }
-    return super.code(code, language, isEscaped);
+
+    const html = super.code(code, language, isEscaped);
+
+    const { example, illegalExample } = metaData;
+    if (!example && !illegalExample) return html;
+
+    const title = example || illegalExample;
+    const className = `${language} ${example ? "example" : "illegal-example"}`;
+    return html.replace(
+      /^<pre[^>]*>/,
```

Comment:
Is this always safe? In other words, is it guaranteed that markdown will never emit `>` inside pre? We know pre can include another HTML elements, so I want to make sure here.

Reply:
The regex will break on cases like `<pre title="foo>">`, but work fine with `<pre title="foo"><code>` or `<pre title="foo">>` (stops at first `>`). As we're replacing content coming from marked, it should be mostly safe.

Reply:
Could you add tests where pre includes:
1. > as a plain text
2. HTML tags

That would ensure this will be safe.

Reply:
Added test. 
Though, even replacing `"<pre>"` would be fine instead of regex, but regex is safer for future changes.

Reply:
Hmm, how about just `<pre>` for now and do regex when we know there is a need? I don't think triple ticks are expected to have any attributes.